### PR TITLE
Issue 4903 rel attribute

### PR DESCRIPTION
**Pull request by Marko64**
- Fixed rel attribute for the text link
- Fixed rel attribute for the button block

Add a button and a text block and make sure `rel` attributes work well. For example, if no attributes are toggled, rel shouldn't be rendered at all. Try toggling one and more attributes and make sure they render correctly.

Fixes #4903 

# How Has This Been Tested?

# Test checklist

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to manage link attributes, including "nofollow," "sponsored," and "ugc" options for enhanced link control.

- **Bug Fixes**
  - Ensured the "rel" attribute is correctly updated or removed based on link settings to maintain valid HTML output.

- **Refactor**
  - Streamlined the handling of link attributes to ensure consistency and accuracy across different components.

### CODE REVIEW

File: src/components/toolbar/components/text-link/utils.js [hunk 1]
```diff
@@ -30,6 +30,11 @@ export const createLinkAttributes = ({
...
 	if (sponsored) attributes.rel += ' sponsored';
 	if (ugc) attributes.rel += ' ugc';
+	if (!noFollow && !sponsored && !ugc) {
+		delete attributes.rel;
+	} else {
+		attributes.rel = attributes.rel.trim();
+	}
 
 	// Clean empty attributes, as it returns error on RichText
 	// and trims the rest
```
[the remark this answers is not in the document]
Comment (Marko64) [reply src/components/toolbar/components/text-link/utils.js:40]: It's a toggle which always retrieve a string, so this sounds unnecessary.

**Comment by elzadj** [comment]
@Marko64 Need to update snapshots, because new generated code doesn't have ```rel=""``` now.

Also - we have empty ```title=""``` too, maybe need to fix it the same way in a new issue? Or we can automatically generate the title and use the url for it, if user did not add a custom link title.

@ckp267 what's better from the point of view of SEO? No title, or title = url by default?

**Comment by Marko64** [comment]
From what I found using Google, empty title is a no, and just repeating information in the title is also a no. That means, title is only used when you want to have something meaningful explaining the link better. I'm not sure what ideally we want to have there automatically - if anything.

**Review by elzadj — CHANGES_REQUESTED** [review]
```
if (!noFollow && !sponsored && !ugc) {
		delete attributes.rel;
	}
```

removes rel even if it's not empty:

Screenshot from 2023-11-29 14-09-39

**Review by elzadj — APPROVED** [review]

**Pull request merged**